ConsoleApplication/ApplicationConfig.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ The MIT License (MIT)
+
+Copyright (c) 2015 Microsoft Corporation
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace DeltaQueryApplication
+{
+    /// <summary>
+    /// Description of the configuration of an AzureAD public client application (desktop/mobile application). This should
+    /// match the application registration done in the Azure portal
+    /// </summary>
+    public class AppConfiguration
+    {
+        /// <summary>
+        /// instance of Azure AD, for example public Azure or a Sovereign cloud (Azure China, Germany, US government, etc ...)
+        /// </summary>
+        public string Instance { get; set; } = "https://login.microsoftonline.com/{0}";
+
+        /// <summary>
+        /// The Tenant is:
+        /// - either the tenant ID of the Azure AD tenant in which this application is registered (a guid)
+        /// or a domain name associated with the tenant
+        /// - or 'organizations' (for a multi-tenant application)
+        /// </summary>
+        public string TenantId { get; set; }
+
+        /// <summary>
+        /// Guid used by the application to uniquely identify itself to Azure AD
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// Scopes needed by the application to work
+        /// </summary>
+        public string [] Scopes { get; set; }
```

Comment:
Didn't you decide to use "https://graph.microsoft.com/.default" ?

Reply:
This has now been resolved [here](https://github.com/microsoftgraph/consoleApp-deltaQuery-dotNet-sample/pull/19/commits/eea56adf250ea654d5a4663f495507fd1961214e)